docs/development/components-and-directories.md

```diff
@@ -0,0 +1,206 @@
+for development purpose.
+
+**`docs`** may also exist in individual components but then only has the scope of
+that component. On the one hand, we need a systemwide entrypoint for documentation
+to look into high level concepts, processes and provide direction to other resources.
+On the other hand, documentation about certain components should be close to the
+component in question,  
+
+**`lang`**, **`template`** and **`Customizing`** stay as is for the time being, as
+the system can only support a certain amount of change per timeframe. In the long
+run it seems advisable to move these folders as well. **`lang`** might be a candidate
+to be split up into single components as well. **`templates`** might be moved
+completely into the UI-framework once legacy UI components are removed completely.
+**`Customizing`** will be a candidate for future treatment once `lang` and `template`
+are reconsidered.
+
+**`public`** is introduced so installations can stop exposing each and every file
+in the ILIAS repository. This is bad for security and reliability, as the ILILAS
+repository contains loads of files that should not be accessible via web: files
+containing automated tests, build artifacts, documentation.
+
+
+### `components` Folder and Namespacing
+
+The `components` folder is the central folder for the ILIAS development. It contains
+components that are maintained and developed via the processes of the ILIAS society
+and community. It may also contain components developed by third parties, formerly
+known as plugins. Hence, this folder is an implementation of one central aim of
+the [Component Revision 2022](https://docu.ilias.de/goto.php?target=wiki_1357_Component_Revision_2022):
+Mechanisms used by core components and plugins should be aligned.
+
+#### How
+
+The top-level in the `components` starts with a list of all vendors that provide
+components for the installation at hand. Hence, for a standard ILIAS, only one
+folder `ILIAS` will be contained. When the installation uses plugins, the vendors
+of the plugin will have folders as well.
+
+The folders of the individual vendors in turn contain one folder for each component
+provided by the vendor in question. Hence a component folder might look as such:
+
+* ILIAS
+  * AccessControl
+  * Administration
+  * Blog
+  * Course
+  * ...
+* ServiceProvider1
+  * Plugin1
+  * Plugin2
+* ServiceProvider2
+  * Plugin1
+
+Each of these folder contains an according component. The `\src` directory
+in that folder contains the root of an according namespace, as described
+by [PSR-4](https://www.php-fig.org/psr/psr-4/). For example: `components/ILIAS/AccessControl`
+contains the component `ILIAS\AccessControl`, `components/ILIAS/AccessControl/src`
+is the root of the namespace `ILIAS\AccessControl`.
+
+A components folder also contains this substructure:
+
+* `\docs` for component specific documentation
+* `\src` for the code to be used for production
+* `\tests for code that performs automated tests
```

Comment:
Should be
```markdown
* `\tests` for code that performs automated tests
```
(add the closing backtick).

And for these three lines: I’d write `tests/` instead of `\tests` to signify they are directories. May be personal preference, but is more common (have not seen the latter out there, but often the former).

Reply:
Just broken fingers due to odd php namespace names :sweat_smile: